app/models/solr_document.rb

```diff
@@ -39,4 +39,33 @@ class SolrDocument
   # and Blacklight::Document::SemanticFields#to_semantic_values
   # Recommendation: Use field names from Dublin Core
   use_extension(Blacklight::Document::DublinCore)
+
+  def iiif_manifest_url
+    if self[:iiif_manifest_ssim].present?
+      follow_redirects(self[:iiif_manifest_ssim].first)
+    end
+  end
+
+  private
+
+    def follow_redirects(url)
+      host = URI.parse(url).host
+
+      # Skip any further redirect business if we already know that we're going to get a
+      # response with good CORS headers.
+      return url if ['cdm17287.contentdm.oclc.org', 'digital.libraries.psu.edu'].include?(host)
```

Comment:
@whereismyjetpack, I left this domain check in here for now. I think that it's a good idea to add a check for a CORS header in the response also - which I'll do.

My reasoning for checking the domain first is that every web request that we perform here is going to be an opportunity for something that's out of our control to go wrong that could make the whole page slow to load or cause it to explode altogether. I'm assuming that a lot of the manifest URLs in the index are at the Content DM domain and served with permissive CORS headers. So for all of those cases, I figured that it would be better to allow the IIIF viewer to handle any errors on the front end so that _only_ the viewer blows up (instead of the whole page).

Do you think that makes sense?

Reply:
I think this makes sense. Since we presently only have 6 of these things, it's not clear if most will be content DM etc. I think it looks good the way it is though.